xds/src/main/java/io/grpc/xds/XdsClusterResource.java

```diff
@@ -282,16 +241,41 @@ private static StructOrError<CdsUpdate.Builder> parseNonAggregateCluster(
           + ": transport-socket-matches not supported.");
     }
     if (cluster.hasTransportSocket()) {
-      if (!TRANSPORT_SOCKET_NAME_TLS.equals(cluster.getTransportSocket().getName())) {
-        return StructOrError.fromError("transport-socket with name "
-            + cluster.getTransportSocket().getName() + " not supported.");
+      String transportSocketName = cluster.getTransportSocket().getName();
+
+      if (!TRANSPORT_SOCKET_NAME_TLS.equals(transportSocketName) &&
+          (isEnabledXdsHttpConnect && !TRANSPORT_SOCKET_NAME_HTTP11_PROXY.equals(transportSocketName))) {
```

Comment:
This logic looks wrong. When `isEnabledXdsHttpConnect == false`, the condition will never be `true`, unlike the old condition. But I don't see any other check for TRANSPORT_SOCKET_NAME_TLS in this method. You also have an unnecessary set of parentheses; I don't know if that is related.